### PR TITLE
major update of version management

### PR DESCRIPTION
**Pull request by grimmjo**
Hey there,

this PR combines the following 5 PRs and makes the obsolete:
- #203 
- #204 
- #207 
- #209 
- #212 

Can you have a quick review on them as I'd like to release a new minor version soon 😄

### CODE REVIEW

File: buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    testRuntimeOnly 'ch.qos.logback:logback-classic:1.2.7'
+}
+
+test {
+    useJUnitPlatform()
+
+    jvmArgs '-Duser.country=US'
+    jvmArgs '-Duser.language=en'
+    failFast = true
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle:20]: :x: this should probably not be pushed as is. On CI you want all tests to be executed.

File: buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    mavenCentral()
+}
+
+dependencies {
+    compileOnly 'com.google.code.findbugs:jsr305' // for "warning: unknown enum constant When.MAYBE"
+
+    testRuntimeOnly 'ch.qos.logback:logback-classic:1.2.7'
+}
+
+test {
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle:15]: In order to preserve laziness, should be `tasks.named("test", Test) { ... }`

File: buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    useJUnitPlatform()
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle:16]: :x: This is redundant

File: buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    jvmArgs '-Duser.country=US'
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle:18]: mmm, if we don't do this already we should think about having that directly in the common plugin

File: settings.gradle [hunk 5]
```diff
@@ -6,12 +6,26 @@ pluginManagement {
...
-rootProject.name = 'nats'
+rootProject.name = 'micronaut-nats'
 
 include 'nats'
 include 'docs-examples:example-groovy'
 include 'docs-examples:example-java'
 include 'docs-examples:example-kotlin'
+
+enableFeaturePreview 'TYPESAFE_PROJECT_ACCESSORS'
+enableFeaturePreview 'VERSION_CATALOGS'
```
Comment (melix) [inline settings.gradle:20]: :x: Using Gradle 7.4 the `VERSION_CATALOGS` feature is no longer preview.

File: build.gradle [hunk 6]
```diff
@@ -1,27 +1,10 @@
...
+    id 'io.micronaut.build.internal.dependency-updates'
+    id 'io.micronaut.build.internal.quality-reporting'
 }
 
-apply plugin: "io.micronaut.build.internal.docs"
-apply plugin: "io.micronaut.build.internal.dependency-updates"
-
-
-def prepareGuideHotfix = tasks.register("prepareGuideHotfix") {
+def prepareGuideHotfix = tasks.register('prepareGuideHotfix') {
```
Comment (melix) [inline build.gradle:7]: :o: Is this still required? The task looks very suspicious. I suggest to run `./gradlew clean`, then `./gradlew docs` and check the output.

**Review by melix — CHANGES_REQUESTED** [review]
That's great cleanup, thanks! I added a few comments, for my own understanding mostly.

File: build.gradle (hunk 6 above)
Comment (grimmjo) [reply build.gradle:7]: If removed the whole block ;)

File: settings.gradle (hunk 5 above)
Comment (grimmjo) [reply settings.gradle]: I've removed the line.

File: buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle (hunk 4 above)
Comment (grimmjo) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle]: If I checked this correctly, the MicronautBuildCommonPlugin already applies those settings.
https://github.com/micronaut-projects/micronaut-build/blob/44d6162a24b98859b58440217c88d339f4a6b772/src/main/groovy/io/micronaut/build/MicronautBuildCommonPlugin.groovy#L112 

Can you double check this, @melix?

File: buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle (hunk 3 above)
Comment (grimmjo) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle]: I've removed the whole testing block as it's redundant.

File: buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle (hunk 4 above)
Comment (melix) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.nats-module.gradle]: You're correct, so the whole block can be removed

**Comment by grimmjo** [comment]
@melix I think the "Static Analysis" Action still needs some credential configuration. Can you configure it?

**Comment by melix** [comment]
This needs support from @alvarosanchez

**Comment by grimmjo** [comment]
thanks @alvarosanchez.

Now just somebody needs to click "merge" 😄

**Comment by melix** [comment]
merged, thanks!

**Pull request merged**